Model/Quote/Checkout.php

```diff
@@ -324,4 +327,20 @@ public function hideDeliveryOptionsForProduct()
 
         return $this;
     }
+
+    /**
+     * @param  bool   $isMailboxPackage
+     * @param  bool   $showPickup
+     * @param  string $carrier
+     *
+     * @return bool
+     */
+    private function isPickupAllowed(bool $isMailboxPackage, bool $showPickup, string $carrier): bool
+    {
+        return ! ($this->package->deliveryOptionsDisabled || ($isMailboxPackage && ! $showPickup))
```

Comment:
Dit is een veel te grote en slecht leesbare condition. Je hebt ook 2 parameters die niet nodig zijn omdat ze alleen hiervoor gebruikt worden. Je kunt dit refactoren naar zoiets, dan houd je alleen de parameter `$carrier` over en kun je wel in een oogopslag door de namen van de variabelen te lezen de logica snappen.

```
$isMailboxPackage     = $this->getPackageType()['packageType'] === self::PACKAGE_TYPE_MAILBOX;
$pickupEnabled        = $this->helper->getBoolConfig($carrier, 'pickup/active');
$showPickupForMailbox = $this->helper->getBoolConfig($carrierPath[$carrier], 'mailbox/pickup_mailbox');
$showPickup           = ! $isMailboxPackage || $showPickupForMailbox;

return ! $this->package->deliveryOptionsDisabled && $pickupEnabled && $showPickup;
```